cdap-ui/app/cdap/components/PreviewData/RecordView/RecordTable.tsx

```diff
@@ -15,53 +15,140 @@
+      msg = T.translate(`${I18N_PREFIX}.noPreviewRunning`, { recordType });
+    }
+    return (
+      <div>
+        <Heading type={HeadingTypes.h3} label={msg} className={classes.messageText} />
+      </div>
+    );
+  };
+
+  const renderList = (visibleNodeCount: number, startNode: number) => {
+    if (!record) {
+      return;
+    }
+    return headers.slice(startNode, startNode + visibleNodeCount).map((fieldName, i) => {
+      const processedFieldName = format(fieldName);
+      const processedValue = format(record[fieldName]);
+      return (
+        <React.Fragment>
+          <Grid
+            container
+            direction="row"
+            wrap="nowrap"
+            className={classnames(classes.row, { oddRow: (i + startNode + 1) % 2 })}
+            key={`gridrow-${i}`}
+          >
+            <Grid
+              item
+              className={classnames(classes.cell, classes.recordCell)}
+              title={processedFieldName}
+            >
+              {processedFieldName}
+            </Grid>
+            <Grid
+              item
+              className={classnames(classes.cell, classes.recordCell)}
+              title={processedValue}
+            >
+              {processedValue}
+            </Grid>
+          </Grid>
+        </React.Fragment>
+      );
+    });
+  };
+
+  // When the selected record number is out of range
+  if (!record) {
+    return noRecordMsg();
+  }
+
   return (
-    <Paper className={classes.root}>
-      <Table>
-        <TableHead>
-          <TableRow className={classes.row}>
-            <CustomTableCell>Field</CustomTableCell>
-            <CustomTableCell>Value</CustomTableCell>
-          </TableRow>
-        </TableHead>
-        <TableBody>
-          {headers.map((fieldName, i) => {
-            return (
-              <TableRow className={classes.row} key={`tr-${i}}`}>
-                <CustomTableCell>{format(fieldName)}</CustomTableCell>
-                <CustomTableCell>{format(record[fieldName])}</CustomTableCell>
-              </TableRow>
-            );
-          })}
-        </TableBody>
-      </Table>
+    <Paper className={classnames(classes.root, classes.recordContainer)}>
+      <Grid container direction="column" wrap="nowrap">
+        <Grid item>
+          <Grid
+            container
+            direction="row"
+            justify="center"
+            alignItems="center"
+            className={classes.headerRow}
+          >
+            <Grid item className={classnames(classes.headerCell, classes.cell, classes.recordCell)}>
+              {T.translate(`${I18N_PREFIX}.fieldName`)}
+            </Grid>
+            <Grid item className={classnames(classes.headerCell, classes.cell, classes.recordCell)}>
+              {T.translate(`${I18N_PREFIX}.value`)}
+            </Grid>
+          </Grid>
+        </Grid>
+        <Grid item>
+          <VirtualScroll
+            itemCount={() => headers.length}
+            visibleChildCount={25}
```

Comment:
Maybe we can let these numbers `25` and `40` be defined with a constant in case we may need to reuse them

Reply:
Extracted as constants!